main.go

```diff
@@ -42,49 +73,39 @@ func main() {
 			panic(err.Error())
 		}
 
-		restartThreshold := int32(NUMBER_RESTARTS)
-
-		for _, item := range pods.Items {
-			StatusLoop:
-			for _, status := range item.Status.ContainerStatuses {
+		for _, pod := range pods.Items {
+		StatusLoop:
+			for _, status := range pod.Status.ContainerStatuses {
 				waiting := status.State.Waiting
 				if waiting != nil {
 					reason := waiting.Reason
-					_, reasonInWaitingReasons := waitingReasons[reason]
-					if reasonInWaitingReasons || (reason == "CrashLoopBackOff" && status.RestartCount > restartThreshold) {
-						if reason == "CrashLoopBackOff" {
-							fmt.Printf("%s / %s has %s restarts, which is over the %s restart limit.", item.Namespace,
-								item.Name, strconv.Itoa(int(status.RestartCount)), strconv.Itoa(int(restartThreshold)))
-						} else if reasonInWaitingReasons {
-							fmt.Printf("%s / %s has a status of %s, which is configured to be deleted.", item.Namespace,
-								item.Name, reason)
-						}
 
-						rs, err := clientset.AppsV1().ReplicaSets(item.Namespace).Get(item.OwnerReferences[0].Name, metav1.GetOptions{})
+					crawlerConfigDetails, ok := waitingReasons[reason]
+					if ok {
```

Comment:
Maybe we can consider changing this "ok" to something more descriptive? Or is "ok" good enough?

Reply:
maybe 'exists'?

Reply:
That would work. Other ones I thought of might be 'included', 'found', 'inside', etc. Probably any of them are fine. Maybe it's just me but 'ok' makes me think 200 OK as in an HTTP response code so interesting it's sort of a Golang pseudo-convention.

Reply:
Looks like Effective Go references it as a the "comma ok" idiom for this purpose, so actually, I think it's best we leave it. For Go-familiar developers, "ok" might actually be the most descriptive.

https://golang.org/doc/effective_go.html#maps

Reply:
Also, actually, as @bhaskar-reddy has mentioned in a different PR and that offset() example in Effective Go shows, I think in this case we can make these two lines into:

`
if crawlerConfigDetails, ok := waitingReasons[reason]; ok {

...
`

Reply:
I made the "ok" one-line change in a commit, so going to "resolve" this conversation. Feel free to un-resolve if there are any objections.